README.md

```diff
@@ -91,13 +91,16 @@ Publish package tarball:
 Unpack package tarball:
 
 ```erlang
-{ok, #{checksum := Checksum, contents := Contents, metadata := Metadata}} = hex_tarball:unpack(Tarball, memory).
+{ok, #{outer_checksum := Checksum, contents := Contents, metadata := Metadata}} = hex_tarball:unpack(Tarball, memory).
 ```
 
+Remember to verify the outer tarball checksum against the registry checksum
+returned from `hex_repo:get_package(Config, Package)`.
+
 Create package tarball:
 
 ```erlang
-{ok, {Tarball, Checksum}} = hex_tarball:create(Metadata, Contents).
+{ok, {Tarball, InnerChecksum, OuterChecksum}} = hex_tarball:create(Metadata, Contents).
```

Comment:
not sure if we'd ever add more fields but in case we do, wdyt about this shape?

```erlang
{ok, {Tarball, #{inner_checksum := InnerChecksum, outer_checksum := OuterChecksum}} =
    hex_tarball:create(Metadata, Contents).
```

or

```erlang
{ok, #{tarball := Tarball, inner_checksum := InnerChecksum, outer_checksum := OuterChecksum}}
```

Perhaps by using map it would be slightly easier to eventually remove inner_checksum (on v1.0 or whatever)?

One thing we briefly discusses adding was returning warnings, which would be another key in the map, although that might have been just for unpacking.

Reply:
The map sounds good, as long as we break people pattern matching on this so that they have to check it if they missed the changelog entries.